examples/cpp-simple/server.cc

```diff
@@ -115,5 +120,9 @@ int main() {
     }
   }
 
+  stop_threads.store(1);
+  health.join();
```

Comment:
There are other "return -1" statements that will return from main without joining these threads. Will that also exhibit the problem this PR is fixing?

Reply:
@aLekSer did you have an answer to this question?

Reply:
Yes, that could lead to an issue, so need one more update.